Sources/OpenAPIRuntime/Interface/Body.swift

```diff
@@ -0,0 +1,535 @@
+        if iterationBehavior == .single {
+            lock.lock()
+            defer {
+                lock.unlock()
+            }
+            guard !locked_iteratorCreated else {
+                fatalError(
+                    "OpenAPIRuntime.Body attempted to create a second iterator, but the underlying sequence is only safe to be iterated once."
+                )
+            }
+            locked_iteratorCreated = true
+        }
+        return sequence.makeAsyncIterator()
+    }
+}
+
+// MARK: - Transforming the body
+
+extension Body {
+
+    /// Creates a body where each chunk is transformed by the provided closure.
+    /// - Parameter transform: A mapping closure.
+    /// - Throws: If a known length was provided to this body at
+    /// creation time, the transform closure must not change the length of
+    /// each chunk.
+    public func mapChunks(
+        _ transform: @escaping @Sendable (Element) async -> Element
+    ) -> Body {
+        let validatedTransform: @Sendable (Element) async -> Element
+        switch length {
+        case .known:
+            validatedTransform = { element in
+                let transformedElement = await transform(element)
+                guard transformedElement.count == element.count else {
+                    fatalError(
+                        "OpenAPIRuntime.Body.mapChunks transform closure attempted to change the length of a chunk in a body which has a total length specified, this is not allowed."
+                    )
+                }
+                return transformedElement
+            }
+        case .unknown:
+            validatedTransform = transform
+        }
+        return Body(
+            sequence: map(validatedTransform),
+            length: length,
+            iterationBehavior: iterationBehavior
+        )
+    }
+}
+
+// MARK: - Consumption utils
+
+extension Body {
+
+    /// An error thrown by the `collect` function when the body contains more
+    /// than the maximum allowed number of bytes.
+    private struct TooManyBytesError: Error, CustomStringConvertible, LocalizedError {
+        let maxBytes: Int
+
+        var description: String {
+            "OpenAPIRuntime.Body contains more than the maximum allowed \(maxBytes) bytes."
+        }
+
+        var errorDescription: String? {
+            description
+        }
+    }
+
+    /// An error thrown by the `collect` function when another iteration of
+    /// the body is not allowed.
+    private struct TooManyIterationsError: Error, CustomStringConvertible, LocalizedError {
+
+        var description: String {
+            "OpenAPIRuntime.Body attempted to create a second iterator, but the underlying sequence is only safe to be iterated once."
+        }
+
+        var errorDescription: String? {
+            description
+        }
+    }
+
+    /// Accumulates the full body in-memory into a single buffer
+    /// up to `maxBytes` and returns it.
```

Comment:
This contradicts each other. Collects the whole body and up to.

Reply:
Agreed the wording needs work. The intention is: you specify the maximum size you're willing to buffer, and if the real body is larger, you get an error thrown (to protect your memory limits).

What terminology would you use here instead?